internal/validations/clusterlogforwarder/inputs/validate.go

```diff
@@ -39,12 +42,11 @@ func Verify(inputs []loggingv1.InputSpec, status *loggingv1.ClusterLogForwarderS
 		case len(status.Inputs[input.Name]) > 0:
 			badInput("duplicate name: %q", input.Name)
 		// Check if inputspec has application, infrastructure, audit or receiver specs
-		case input.Application == nil && input.Infrastructure == nil && input.Audit == nil && input.Receiver == nil:
-			badInput("inputspec must define one or more of application, infrastructure, audit or receiver")
-		case input.HasPolicy() && input.Application.ContainerLimit != nil && input.Application.GroupLimit != nil:
-			badInput("inputspec must define only one of container or group limit")
-		case input.HasPolicy() && input.GetMaxRecordsPerSecond() < 0:
-			badInput("inputspec cannot have a negative limit threshold")
+		case !hasOneType(input):
+			badInput("inputspec must define one and only one of: application, infrastructure, audit or receiver")
+		case !validApplication(input, status, extras):
```

Comment:
Do you need to set status on these checks, with badInput() as well?

Reply:
`badInput` is a legacy convenience mechanism for setting status which IMO does not properly create conditions.  The various functions set a status which should more correctly reflect a specific issue.